jenkins/databricks/test.sh

```diff
@@ -15,8 +15,33 @@
 # limitations under the License.
 #
 
+# This script sets the  environment to run tests of RAPIDS Accelerator for Apache Spark on DB.
+# All the environments can be overwritten by shell variables:
+#   LOCAL_JAR_PATH: Location of the RAPIDS jars
+#   SPARK_CONF: Spark configuration parameters
+#   BASE_SPARK_VERSION: Spark version [3.1.2, 3.2.1, 3.3.0]. Default is pulled from current instance.
+#   ICEBERG_VERSION: The iceberg version. To find the list of supported ICEBERG versions,
+#                    check https://iceberg.apache.org/multi-engine-support/#apache-spark
+#   ICEBERG_SCALA_VERSION: Scala version used by ICEBERG jars. Default is 2.12.
```

Comment:
I'm confused why this is called out explicitly.  How could we support pulling in a Scala version for Iceberg that doesn't match the Scala version used for all other jars?  If it's truly necessary, why are we not doing it for other jars with the Scala version in the artifact name?  Either this should be removed or a comment should be added explaining why this special case handling is necessary rather than using a consistent Scala version across all artifacts with a Scala version in the artifact name (e.g.: something like SCALA_VERSION used in jenkins/databricks/build.sh and jenkins/databricks/deploy.sh)

Reply:
- I fixed the SCALA version to be consistent. 
- I filed https://github.com/NVIDIA/spark-rapids/issues/7210 as followup to install the Iceberg jars provided by databricks in `jenkins/databricks.build.sh`
- I added a note in `docs/additional-functionality/iceberg-support.md` that the plugin does not support iceberg in databricks
- I clarified a comment in jenkins/spark-tests.sh that plugin is not supporting iceberg with spark3.3+